includes/admin_import.php

```diff
@@ -155,47 +155,133 @@ function tsml_import_page()
             }
         }
 
-        //add data source
+
+        //add data source  
         if (!empty($_POST['tsml_add_data_source']) && isset($_POST['tsml_nonce']) && wp_verify_nonce($_POST['tsml_nonce'], $tsml_nonce)) {
 
-            //sanitize URL, name, parent region id, and Change Detection values
+            //initialize variables 
+            $is_google_sheets_import = false;
+            $import_updates = $db_ids_to_delete = $message_lines = [];
+            $tbl_col1_txt = __('Update Mode', '12-step-meeting-list');
+            $tbl_col2_txt = __('Meeting Name', '12-step-meeting-list');
+            $tbl_col3_txt = __('Day of Week', '12-step-meeting-list');
+            $tbl_col4_txt = __('Meeting Time', '12-step-meeting-list');
+            $tbl_col5_txt = __('Last Updated', '12-step-meeting-list');
+
+            //sanitize URL, name, and parent region id values
             $data_source_url = trim(esc_url_raw($_POST['tsml_add_data_source'], array('http', 'https')));
             $data_source_name = sanitize_text_field($_POST['tsml_add_data_source_name']);
             $data_source_parent_region_id = (int) $_POST['tsml_add_data_source_parent_region_id'];
-            $data_source_change_detect = sanitize_text_field($_POST['tsml_add_data_source_change_detect']);
 
             //try fetching	
             $response = wp_safe_remote_get($data_source_url, [
                 'timeout' => 30,
                 'sslverify' => false,
             ]);
-            if (is_array($response) && !empty($response['body']) && ($body = json_decode($response['body'], true))) {
 
-                //if already set, hard refresh
-                if (array_key_exists($data_source_url, $tsml_data_sources)) {
-                    tsml_delete(tsml_get_data_source_ids($data_source_url));
+            //set response body to an empty array rather than erroring
+            $body = (is_array($response) ? json_decode($response['body'], true) : []);
+
+            //allow reformatting as necessary
+            if (strpos($data_source_url, "sheets.googleapis.com") !== false) {
+                $meetings = tsml_import_reformat_googlesheet($body);
+                $is_google_sheets_import = true;
+            } elseif (function_exists('tsml_import_reformat')) {
```

Comment:
don't think this should be an `elseif` - we currently run tsml_import_reformat after tsml_import_reformat_googlesheet if it exists

Reply:
Good to know. We will make it a simple if() statement.